### ISSUE TITLE
Consistent handling of addresses

### ISSUE THREAD

**Issue by larsks** [centos-opstools/opstools-ansible#59]
I would like to propose that for all of our services, we define the following variables:

    <servicename>_port:
    <servicename>_listen_addresses:
    <servicename>_server_address:

A service binds to addresses in `<servicename>_listen_addresses` on `<servicename>_port`.  Clients of the service connect to `<servicename>_server_address` (on `<servicename>_port`).  So instead of e.g., the senu module having `sensu_redis_server`, the `sensu` module would depend on the `redis` module and would use `redis_server_address`.

This will in general require separating role defaults from the actual implementation (so, having both a `redis` role that sets the defaults, and a `redis/server` role that performs the actual configuration)

**Comment by larsks** [comment]
Of course, some services are only able to bind to a single address.  I don't know if we aim for consistency (use a single-element list for `<servicename>_listen_addresses), or alternately just have a `<servicename>_listen_address` (singular) for that situation.

**Comment by paramite** [comment]
+1

**Comment by paramite** [comment]
I would prefer *_listen_address which accepts list for services where it make sense.

**Comment by larsks** [comment]
Yeah.  And when we do all this, we're going to need to update our firewall handling to support specific addresses, rather than globally opening ports.
